metadata-ingestion/src/datahub/utilities/urns/_urn_base.py

```diff
@@ -0,0 +1,228 @@
+import functools
+import urllib.parse
+from abc import abstractmethod
+from typing import ClassVar, Dict, List, Optional, Type, TypeVar
+
+from deprecated import deprecated
+
+from datahub.utilities.urns.error import InvalidUrnError
+
+URN_TYPES: Dict[str, Type["_SpecificUrn"]] = {}
+
+
+def _split_entity_id(entity_id: str) -> List[str]:
```

Comment:
Some doctests / examples of what this is doing would be nice. I eventually got it but took a sec. Also kinda odd to me that it basically ignores parentheses, besides checking that there's the right amount of them in an appropriate order.

It's less efficient but think it'd be clearer to break this up:
```
PARENS = ["(", ")"]
def _split_entity_id(entity_id: str) -> List[str]:
    if not (entity_id.startswith("(") and entity_id.endswith(")")):
        return [entity_id]
    parens = [v for v in entity_id if v == "(" or v == ")"]
    if not _parens_valid(parens):
        raise InvalidUrnError(...)
    return entity_id[1:-1].split(",")
```
I think that's what this function is doing?

Reply:
this was copied from the existing urn code, so I'm inclined to leave it as is since it seems to work